examples/querying_entity_versions.ipynb

```diff
@@ -0,0 +1,782 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "82c3962a-27b5-4375-b4fa-dbd14dcfc93b",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "> **Result:**\n",
+       "> `<openassetio._openassetio.EntityReference object at 0x113c8d0f0>`"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "logo_ref = manager.createEntityReference(\"bal:///project_artwork/logos/openassetio\")\n",
+    "\n",
+    "helpers.display_result(repr(logo_ref))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ec69ae04-1895-4243-a949-71360175b173",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "Now we have an entity reference for our logo, we can use the API to learn more about it.\n",
+    "\n",
+    "Versioning information is [not part of the core API](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/doc/decisions/DR017-Entity-version-queries.md) itself. MediaCreation traits that describe how we generally work with versions in the post-production industry are used with the core API methods."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4348da46-9467-4734-9f96-d84e26f7c11f",
+   "metadata": {
+    "editable": true,
+    "jp-MarkdownHeadingCollapsed": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Querying an entity's version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5e32a587-5233-441e-9b80-1b2d6d3769fb",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "An entity's version can be resolved using the imaginatively named `VersionTrait` from the `lifecycle` module of the `openassetio_mediacreation` package `traits`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "0fcf0214-2561-488a-8eee-0330ca27710d",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from openassetio_mediacreation.traits.lifecycle import VersionTrait\n",
+    "from openassetio.access import ResolveAccess\n",
+    "\n",
+    "entity_data = manager.resolve(logo_ref, {VersionTrait.kId}, ResolveAccess.kRead, context)\n",
+    "\n",
+    "version_trait = VersionTrait(entity_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6c4f0ff-0e5d-40b7-84d6-d46efa8cba7a",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "We use the `VersionTrait` class as a wrapper around the generic `TraitsData` returned from the API to inspect it's properties - which contain information about the entity's version. This helps ensure we don't accidentally mis-spell traits or their properties.\n",
```

Comment:
[Consider] I think this is too specific, and sort of undermines the trait wrapper by talking too in-detail about `TraitsData`. What about?:

```cpp
    "`VersionTrait` is used to wrap the untyped `TraitsData` returned from `resolve`, providing accessors to the entities properties, in this case accessors concerning the entities version information.
```

Reply:
Hows [this](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/59/commits/88c28d72cfb6d60d9b28086485eca6da9cf2adc4)? I suck with generic vs untyped as technically it is typed, just missing concrete accessors.

Reply:
Aye. :+1: